pkg/prometheus/operator.go

```diff
@@ -1428,6 +1446,47 @@ func (c *Operator) selectServiceMonitors(p *monitoringv1.Prometheus) (map[string
 		})
 	}
 
+	// If denied by Prometheus spec, filter out all service monitors that access
+	// the file system.
+	if p.Spec.FilesystemAccessThroughServiceMonitors.Deny {
```

Comment:
this should likely be recorded in the ServiceMonitor's status (I realize this doesn't exist today)

Reply:
Reporting status back into a service monitor seems to be not specific to this pull request. Thus I suggest achieving the above in a separate pull request.

Reply:
Sounds good to me.